experiments/process_sandbox/README.md

```diff
@@ -0,0 +1,36 @@
+Process-based sandbox experiment
+================================
+
+*WARNING*: This is not a complete implementation and almost certainly has security issues.
+
+This is a prototype of loading a library in a sandboxed child process.
+Parts of this are intended to be useable for process-based sandboxing of legacy libraries in Verona programs.
+
+For each sandbox instance, the parent process creates two anonymous shared memory objects.
+One object is large and will be used for the sandbox's heap.
+The large object is mapped at the same address in both the parent and a child process, allowing intra-sandbox pointers to have the same values in both.
+This is mapped read-write in both processes.
+
+The smaller object is a single page, used for the snmalloc page map that covers the large region.
+This is mapped read-write in the parent over the region of the pagemap that contains metadata for the large region.
+It is mapped read-only in the child.
+
+The parent has an allocator associated with the sandbox that allows allocation in the sandbox's heap.
+The parent allocator's message queue is in the shared region, but the rest of the state is not.
```

Comment:
The shared region is writable by the child, wouldn't that be a potential problem?

Reply:
In theory, yes.  That's why we have (had? @mjp41, did we publish it?) code in snmalloc to validate those pointers.  The worst that should be able to happen is that malicious code can trigger a UAF, but since any data in the sandbox is intrinsically untrusted this doesn't affect the threat model.

Reply:
I think at the moment it is "had thought about". I reviewed a lot of places it would be required, but did not implement it.